### PR TITLE
Preprocess and precompute with multiprocessing

### PR DESCRIPTION
**Pull request by dorinapetra**
- use general precompute in script
- preprocess and precompute at the same time
- preprocess and precompute can use multiprocessing

### CODE REVIEW

**Review by attilanagy234 — CHANGES_REQUESTED** [review]
Great job 🔥 All looks good, but could we add a test for the preprocess_and_precompute method? Using some dummy data is totally fine, just to see that both the preprocessing and the dep tree precomputation works.

**Review by Baaart25 — APPROVED** [review]
I found some nice to fix things

File: src/hu_nmt/test/preprocessor/preprocessor_test.py [hunk 1]
```diff
@@ -26,22 +28,6 @@ def test_is_correct_language_false(self):
 
         self.assertFalse(self.preprocessor.is_correct_language(source_sentence, target_sentence))
 
-    def test_contains_one_sentence_true(self):
```
[the remark this answers is not in the document]
Comment (dorinapetra) [reply src/hu_nmt/test/preprocessor/preprocessor_test.py:29]: I restored them

**Review by attilanagy234 — APPROVED** [review]
Looking good now, thanks for making the changes

**Pull request merged**